src/actions/bumpThoughtDown.ts

```diff
@@ -63,6 +68,14 @@ const bumpThoughtDown = (state: State, { simplePath }: { simplePath?: SimplePath
       })
     },
 
+    state => {
+      // sort the new thought if there is a sort preference
+      if (sortId) return sort(state, head(simplePath))
```

Comment:
This works, though it's not the best approach. Since the destination context is already fully sorted, we only need to insert the bumped thought into the correct place, rather than re-sort the entire list. While the front-end performance is likely unaffected, generating new ranks for all siblings results in unnecessary changes pushed to the db and thus unnecessary bandwidth.

See the documentation for [rank](https://github.com/cybersemics/em/wiki/Docs#rank) for details on how a value can be inserted without changing the rank of its siblings.

You might find [getSortedRank](https://github.com/cybersemics/em/blob/f545b318167c1f61b4af54c5d01fc35c22b231d9/src/selectors/getSortedRank.ts) to be helpful for this task.

Reply:
`getSortedRank` is what I needed, thanks for the headsup!